internal/testrunner/coverage.go

```diff
@@ -0,0 +1,229 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package testrunner
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/elastic/elastic-package/internal/files"
+)
+
+// GenerateBasePackageCoverageReport generates a coverage report where all files under the root path are
+// marked as not covered. It ignores files under _dev directories.
+func GenerateBasePackageCoverageReport(pkgName, rootPath, format string) (CoverageReport, error) {
```

Comment:
Could this function solved also this issue for pipeline tests ? 
https://github.com/elastic/elastic-package/issues/1624

If all files present in the package are marked as uncovered, there will be coverage (zero) for those files without tests.

Just one concern regarding also about pipelines, if there is a coverage report created by pipeline in some package. With this change there would be lines marked as uncovered (that previously there was no information), that could cause a decrease in the coverage (%). Would that be an issue?

Reply:
> Could this function solved also this issue for pipeline tests ?
> #1624

Probably yes. Tests will generate coverage for all files now, even if it is empty.

> If all files present in the package are marked as uncovered, there will be coverage (zero) for those files without tests.

Correct, if there are no tests, there is no coverage.

> Just one concern regarding also about pipelines, if there is a coverage report created by pipeline in some package. With this change there would be lines marked as uncovered (that previously there was no information), that could cause a decrease in the coverage (%). Would that be an issue?

Yes, there can be a decrease in coverage in several cases. I wouldn't consider this as an issue. Before there was more coverage % just because we were not reporting coverage for many files. We will probably revisit what files we mark as covered or uncovered, and how.